backend/assignments/1/src/main/java/org/example/ExecuteTransaction.java

```diff
@@ -0,0 +1,121 @@
+package org.example;
+
+import java.security.SecureRandom;
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Handles the execution of different types of transactions (BUY, SELL, UPDATE_PRICE, ADD_VOLUME).
+ * Implements the Runnable interface for concurrent execution by threads.
+ */
+public class ExecuteTransaction implements Runnable {
+
+    private final Trader trader;
+    private final Transaction transaction;  // Updated to use Transaction class
+    private final CountDownLatch latch;
+
+    /**
+     * Initializes the ExecuteTransaction with required data.
+     *
+     * @param trader      The trader involved in the transaction.
+     * @param transaction The transaction object.
+     * @param latch       The CountDownLatch for synchronization.
+     */
+    public ExecuteTransaction(Trader trader, Transaction transaction, CountDownLatch latch) {
+        this.trader = trader;
+        this.transaction = transaction;
+        this.latch = latch;
+    }
+
+    /**
+     * Executes the transaction based on its type.
+     */
+    @Override
+    public void run() {
+        try {
+            // Your transaction processing logic goes here
+            String transactionHash = getBlockHash(); // Generate a random transaction hash
+
+            switch (transaction.getType()) {
+                case BUY:
+                    executeBuyTransaction(transaction, transactionHash);
+                    break;
+                case SELL:
+                    executeSellTransaction(transaction, transactionHash);
+                    break;
+                case UPDATE_PRICE:
+                    executeUpdatePriceTransaction(transaction, transactionHash);
+                    break;
+                case ADD_VOLUME:
+                    executeAddVolumeTransaction(transaction, transactionHash);
+                    break;
+                default:
+                    Logging.getMsg().error("Unknown transaction type: {}", transaction.getType());
+            }
+        } finally {
+            // Count down the latch to signal completion
+            latch.countDown();
+        }
+    }
+
+    private void executeBuyTransaction(Transaction transaction, String transactionHash) {
+        if (trader.getBalance() >= transaction.getCoin().getCurrentPrice() * transaction.getQuantity()) {
```

Comment:
Add input validations and null checks for all methods. And abstract them out.

Wrap them in a validator wrapper/function.
What if these checks are general and reusable?